specifyweb/frontend/js_src/lib/components/Statistics/hooks.tsx

```diff
@@ -71,54 +72,56 @@ function backEndStatPromiseGenerator(
+  layout: StatLayout | undefined,
+  defaultLayout: RA<StatLayout> | undefined
+): RA<StatLayout> | undefined {
+  if (layout === undefined || defaultLayout === undefined) {
+    return undefined;
+  }
+  const listToUse = layout.categories.flatMap(({ items }) =>
+    items.filter((item): item is DefaultStat => item.type === 'DefaultStat')
+  );
+  let statNotFound = false;
+  const defaultLayoutFlagged = defaultLayout.map((defaultLayoutPage) => ({
+    label: defaultLayoutPage.label,
+    lastUpdated: undefined,
+    categories: defaultLayoutPage.categories.map(({ label, items }) => ({
+      label,
+      items: items.map((defaultItem) => {
+        const defaultStatNotFound =
+          defaultItem.type === 'DefaultStat' &&
+          !listToUse.some(
+            ({ pageName, categoryName, itemName, pathToValue }) =>
+              pageName === defaultItem.pageName &&
+              categoryName === defaultItem.categoryName &&
+              itemName === defaultItem.itemName &&
+              pathToValue === defaultItem.pathToValue
+          );
+        statNotFound ||= defaultStatNotFound;
+        return {
+          ...defaultItem,
+          isVisible: defaultStatNotFound ? undefined : false,
+        };
+      }),
+    })),
+  }));
+  return statNotFound ? defaultLayoutFlagged : undefined;
+}
 export function useDefaultStatsToAdd(
   layout: StatLayout | undefined,
   defaultLayout: RA<StatLayout> | undefined
 ): RA<StatLayout> | undefined {
-  return React.useMemo((): RA<StatLayout> | undefined => {
-    if (
-      layout === undefined ||
-      defaultLayout === undefined ||
-      layout.categories === undefined
-    ) {
-      return undefined;
-    }
-    const listToUse = layout.categories.flatMap(({ items = [] }) =>
-      items.filter((item): item is DefaultStat => item.type === 'DefaultStat')
-    );
-    let statNotFound = false;
-    const defaultLayoutFlagged = defaultLayout.map((defaultLayoutPage) => ({
-      label: defaultLayoutPage.label,
-      lastUpdated: undefined,
-      categories: defaultLayoutPage.categories.map(({ label, items }) => ({
-        label,
-        items: items.map((defaultItem) => {
-          const defaultStatNotFound =
-            defaultItem.type === 'DefaultStat' &&
-            !listToUse.some(
-              ({ pageName, categoryName, itemName, pathToValue }) =>
-                pageName === defaultItem.pageName &&
-                categoryName === defaultItem.categoryName &&
-                itemName === defaultItem.itemName &&
-                pathToValue === defaultItem.pathToValue
-            );
-          statNotFound ||= defaultStatNotFound;
-          return {
-            ...defaultItem,
-            isVisible: defaultStatNotFound ? undefined : false,
-          };
-        }),
-      })),
-    }));
-    return statNotFound ? defaultLayoutFlagged : undefined;
-  }, [layout, defaultLayout]);
+  return React.useMemo(
+    (): RA<StatLayout> | undefined =>
```

Comment:
(optional) you don't need to specify the return type of the useMemo as it's infered infered automatically from return type of useDefaultStatsToAdd